src/3d/qgs3daxis.h

```diff
@@ -0,0 +1,218 @@
+#include <Qt3DRender/QGeometryRenderer>
+
+#define SIP_NO_FILE
+
+class QgsCameraController;
+class Qgs3DMapSettings;
+class QgsCoordinateReferenceSystem;
+
+/**
+ * \ingroup 3d
+ * \brief Display 3D ortho axis in the main 3D view.
+ *
+ * Axis are displayed in a dedicated viewport which can be placed all around the main viewport.
+ * Axis labels are displayed in a dedicated viewport with a specific camera to act as a billboarding layer.
+ * Axis labels can be changed with the mode combo box from the navigation widget.
+
+ * \note Not available in Python bindings
+ *
+ * \since QGIS 3.26
+ */
+class _3D_EXPORT Qgs3DAxis : public QObject
+{
+    Q_OBJECT
+  public:
+
+    /**
+     * \brief Defaul Qgs3DAxis constructor
+     * \param parentWindow qt3d windows
+     * @param parent3DScene root entity to set as parent
+     * @param camera camera controller used to track camera movements
+     * @param map 3D map settings
+     */
+    Qgs3DAxis( Qt3DExtras::Qt3DWindow *parentWindow,  Qt3DCore::QEntity *parent3DScene, QgsCameraController *camera, const Qgs3DMapSettings *map );
+
+    /**
+     * \brief The Axis enum
+     */
+    enum class Axis
+    {
+      X = 1,
+      Y = 2,
+      Z = 3
+    };
+    Q_ENUM( Axis )
+
+    /**
+     * \brief The AxisViewportPosition enum
+     */
+    enum class AxisViewportPosition
+    {
+      //! top or left
+      Begin = 1,
+      Middle = 2,
+      //! bottom or right
+      End = 3
+    };
+    Q_ENUM( AxisViewportPosition )
+
+    /**
+     * \brief The Mode enum
+     */
+    enum class Mode
+    {
+      //! disabled
+      Off = 1,
+      //! CRS specific.
+      Crs = 2,
+      //! Cube with label
+      Cube = 3
+    };
+    Q_ENUM( Mode )
+
+    /**
+     * \brief set axis representation mode
+     * \param axisMode new node
+     */
+    void setMode( Qgs3DAxis::Mode axisMode );
+
+    /**
+     * \brief set axis viewport position parameters
+     * \param axisViewportSize height/width size in pixel
+     * @param axisViewportVertPos start vertical position
+     * @param axisViewportHorizPos start horizontal position
+     */
+    void setAxisViewportPosition( int axisViewportSize, AxisViewportPosition axisViewportVertPos, AxisViewportPosition axisViewportHorizPos );
+
+    /**
+     * \brief get axis viewport size
+     */
+    int axisViewportSize() const { return mAxisViewportSize;}
+
+    /**
+     * \brief get axis viewport horizontal position
+     */
+    AxisViewportPosition axisViewportHorizontalPosition() const { return mAxisViewportHorizPos;}
+
+    /**
+     * \brief get axis viewport vertical position
+     */
+    AxisViewportPosition axisViewportVerticalPosition() const { return mAxisViewportVertPos;}
+
+    /**
+     * \brief project a 3D position from sourceCamera (in sourceViewport) to a 2D position for destCamera (in destViewport). destCamera and the destViewport act as a billboarding layer. The labels displayed by this process will always face the camera.
```

Comment:
I believe you don't really need to billboard the label 3D objects. 
Alternatively you can match the rotation of the label to face the camera.

Reply:
I have made a previous version with the label rotation corrected according to the camera position but it was not as clean as this version.